public/components/bulkReindexes.react.js

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import { Panel, Button, Table } from 'react-bootstrap';
+import RunningReindex from './runningReindex.react';
+
+
+export default class BulkReindex extends React.Component {
+
+    constructor(props) {
+        super(props);
+
+        this.cancelReindex = this.cancelReindex.bind(this);
+        this.renderCancelButton = this.renderCancelButton.bind(this);
+    }
+
+
+    cancelReindex(reindex, isRunningJob) {
+        this.props.onCancelReindex(reindex, isRunningJob);
+    }
+
+    renderCancelButton(reindex, isRunningJob) {
+        if (isRunningJob && !reindex.settings.supportsCancelReindex) {
+            return (null);
+        } else {
+            return (
+                <Button bsStyle="danger" className="pull-right" type="button" onClick={this.cancelReindex.bind(this, reindex, isRunningJob)}>Cancel</Button>
+            );
+        }
+    }
+
+    render () {
+
+        const completedJobsNodes = this.props.completedReindexes.map(completedJob => {
+            return (
+                <tr key={completedJob.startTime}>
+                    <td>{completedJob.name}</td>
+                    <td>{completedJob.env}</td>
+                    <td>{completedJob.status}</td>
+                    <td>{new Date(completedJob.startTime).toUTCString()}</td>
+                    <td>{new Date(completedJob.finishTime).toUTCString()}</td>
+                </tr>
+            );
+        });
+
+        const runningJobsNodes = this.props.runningReindexes.map(runningJob => {
+            return (
+                <tr key={runningJob.startTime}>
+                    <td>{runningJob.name}</td>
+                    <td>{runningJob.env}</td>
+                    <td>
+                        <RunningReindex documentsIndexes={runningJob.documentsIndexed}
+                                        documentsExpected={runningJob.documentsExpected}
+                                        environment={runningJob.env}
+                                        id={runningJob.id}
+                                        startTime={runningJob.startTime}
+                                        reindex={runningJob}
+                                        isCancelSupported={false}
+                                        onCancelReindex={this.cancelReindex.bind(this)}
```

Comment:
So I ended up doing a horrible hacky workaround here. I set `isCancelSupported` to be false as I wanted to render my cancel button within `BulkReindex` component rather than the `RunningReindex` component. This is because the `RunningReindex` component is largely built to render a cancel button for the only running reindex at that point in time, whereas we wanted to use it for both `pending` and `running` jobs and I wanted to be able to control the placing of the button to keep it consistent.

So with that in mind, the `onCancelReindex` props don't actually do anything here. I'm wondering if there's  a smarter workaround for the rendering of the cancel button, or what I could pass in as a dummy value to the `onCancelReindex` props. 🤔 

Would welcome your opinion @regiskuckaertz !